pkg/api/types.go

```diff
@@ -2766,6 +2767,22 @@ type Preconditions struct {
 	UID *types.UID
 }
 
+// DeletePropagationPolicy decides whether and how garbage collection will be performed.
+type DeletePropagationPolicy string
+
+const (
+	// The default depends on the existing finalizers on the object and the type of the object.
+	DeletePropagationDefault DeletePropagationPolicy = "DeletePropagationDefault"
```

Comment:
`"DeletePropagationDefault"` -- I think the named strings don't need to include "DeletionPropagation", so this can just be `"Default"`. Same for others.